cmd/quadlet/main.go

```diff
@@ -210,6 +211,71 @@ func enableServiceFile(outputPath string, service *parser.UnitFile) {
 	}
 }
 
+func isImageID(imageName string) bool {
+	// All sha25:... names are assumed by podman to be fully specified
+	if strings.HasPrefix(imageName, "sha256:") {
+		return true
+	}
+
+	// However, podman also accepts image ids as pure hex strings,
+	// but only those of length 64 are unambigous image ids
+	if len(imageName) != 64 {
+		return false
+	}
+
+	for _, c := range imageName {
+		if !unicode.Is(unicode.Hex_Digit, c) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func isUnambiguousName(imageName string) bool {
```

Comment:
Can we rename to `isShortName`?  That's the term we've been using so far and it's surprisingly hard to wrap my head around the {un}ambiguous and it being used negated.

Reply:
I got complaints of it not being technically true as it also accepts image ids, which are not short names.

Reply:
Ah, sorry for bikeshedding.  Would `IsFQNorID` work?

Reply:
I don't care, as long as it is acceptable by others.

Reply:
@mtrmac opinions?

Reply:
Why would't isShortName work, it would return false if it is a FQN or an ID?

Reply:
\*shrug\* I’m a stupid person, usually busy, and therefore with insufficient attention span. If you ask _me_ what I want, please make it easy for me to understand what code means, without having to read much of it.

That’s, IMHO, best achieved by naming the function so clearly that it doesn’t need documentation (extra credit: but documenting it anyway). Second best is clearly documenting the function.

The old GNU coding standard of every function having a comment spelling out all parameters and return values, so that people who modify the callers can usually read that comment and don’t have to visit the implementation of the callee at all, continues to seem like a good idea to me, but I realize that’s not a widespread practice.

---

Looking at the current shape of the code, maybe split `isImageID` and `is[NamedReferenceA]ShortName` as two separate helpers, and have `warnIfAmbigiousName` use `if !isImageID && !isNamedReferenceAShortName`?

That would avoid the need to invent a good name for the current, somewhat multi-purpose, function.

Reply:
isNamedReferenceAShortName() implies it returns false for an invalid name string though (like the c/i IsShortName() does), but this doesn't do that.

Reply:
I read `isNamedReferenceAShortName` to mean “if it is not a named reference, the behavior is undefined” (it’s not `isANamedReferenceShortName`.

---

Ultimately, I guess I should not be making that huge a deal out of this? I’m not a core Podman maintainer. I have opinions, but not that high a likelihood of having to actually maintain the code in the future.

So, whatever others decide is fine with me.